framework/src/modules/chain/dpos/delegates_list.js

```diff
@@ -45,12 +45,16 @@ class DelegatesList {
 	}
 
 	/**
-	 * Get shuffled list of active delegate public keys for a specific round -> forger public keys
+	 * Get shuffled list of active delegate public keys (forger public keys) for a specific round.
+	 * The list of delegates used is the one computed at the end of the round `r - delegateListRoundOffset`
 	 * @param {number} round
+	 * @param {number} delegateListRoundOffset
 	 */
-	async getForgerPublicKeysForRound(round) {
+	async getForgerPublicKeysForRound(round, delegateListRoundOffset) {
+		// Delegate list is generated from round 1 hence `roundWithOffset` can't be less than 1
+		const roundWithOffset = Math.max(round - delegateListRoundOffset, 1);
```

Comment:
Do you think adding a check against negative offset values make sense? For instance, if round: 6 and I provide offset: -2 I can get the list for round 8. Just saying..

Reply:
That's a good point but I would prefer doing this in another ticket to avoid more changes and tests broken in this PR 😬